### PR TITLE
GOES event list example

### LINKED ISSUE #3942: Write example showing how to get GOES event list
As discussed at the developer meeting, `get_goes_event_list` should be deprecated and removed, and a gallery example should be added showing how to get the event list using HEK.

### PR DESCRIPTION
**Pull request by hayesla**
### Description
This is an example of how to use the HEK client to query the GOES flare event list between two times, define a filter of GOES class to search for, and save the results as a csv file.

Fixes #3942 

It also means we can remove the `sunpy.instr.goes.get_goes_event_list()`

### CODE REVIEW

File: examples/acquiring_data/querying_the_GOESeventlist.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+print(len(result))
+print(result)
+print(result.keys())
+
+###################################################################
+# We can also specify what GOES class flares we want to query.
+# For example, let's search only for large solar flares with
+# a GOES class > M1.0. This can be achieved by using the FL.GOESCls
+# attribute of the HEK client:
+goes_class_filter = 'M1.0'
```
Comment (wtbarnes) [inline examples/acquiring_data/querying_the_GOESeventlist.py:44]: Is there a reason to make this a separate variable?

File: examples/acquiring_data/querying_the_GOESeventlist.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+###################################################################
+# We then use the client to query the HEK for a list of events
+# that were detected by the GOES X-ray Sensor (XRS) instrument between
+# tstart and tend.
+result = client.search(hek.attrs.Time(tstart, tend),
+                       hek.attrs.EventType(event_type),
+                       hek.attrs.OBS.Observatory == 'GOES')
+
+###################################################################
+# The result is returned as a `Table <sunpy.net.hek.hek.HEKTable>`.
```
Comment (wtbarnes) [inline examples/acquiring_data/querying_the_GOESeventlist.py:31]:
```suggestion
# The result is returned as a `~sunpy.net.hek.hek.HEKTable`.
```

File: examples/acquiring_data/querying_the_GOESeventlist.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+result_m1 = client.search(hek.attrs.Time(tstart, tend),
+                          hek.attrs.EventType(event_type),
+                          hek.attrs.FL.GOESCls > goes_class_filter,
+                          hek.attrs.OBS.Observatory == 'GOES')
+
+print(result_m1)
+
+###################################################################
+# The results returned to the `Table <sunpy.net.hek.hek.HEKTable>`
```
Comment (wtbarnes) [inline examples/acquiring_data/querying_the_GOESeventlist.py:54]:
```suggestion
# The results returned to the `~sunpy.net.hek.hek.HEKTable`
```

File: examples/acquiring_data/querying_the_GOESeventlist.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+# contain alot of information and we may only want to keep some main
```
Comment (wtbarnes) [inline examples/acquiring_data/querying_the_GOESeventlist.py:55]:
```suggestion
# contain a lot of information and we may only want to keep some main
```

File: examples/acquiring_data/querying_the_GOESeventlist.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+# results such as start time, end time, peak time, GOES-class, active
```
Comment (wtbarnes) [inline examples/acquiring_data/querying_the_GOESeventlist.py:56]:
```suggestion
# results such as start time, end time, peak time, GOES-class, and active
```

File: examples/acquiring_data/querying_the_GOESeventlist.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+# region number. This can be done as so:
+new_table = result_m1[['event_starttime', 'event_peaktime',
+                       'event_endtime', 'fl_goescls', 'ar_noaanum', ]]
+
+###################################################################
+# These results can then be saved to a csv file, or whatever file
+# format that `astropy.table.Table` supports
```
Comment (wtbarnes) [inline examples/acquiring_data/querying_the_GOESeventlist.py:63]:
```suggestion
# These results can then be saved to a CSV file or any other file
# format that `~astropy.table.Table` supports
```

**Comment by wtbarnes** [comment]
The docs builds are failing because Circle seems to be installing astropy 4.0.1 (which doesn't play nice with reproject) despite the fact that we are pinning to !=4.0.1... 🤦‍♀

File: examples/acquiring_data/querying_the_GOESeventlist.py [hunk 7]
```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
```
Comment (nabobalis) [inline examples/acquiring_data/querying_the_GOESeventlist.py:1]: What about the filename `querying_the_GOES_event_list.py `?

Comment (hayesla) [reply examples/acquiring_data/querying_the_GOESeventlist.py]: yep - done!

**Pull request merged**